netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -0,0 +1,145 @@
+package org.nlogo.app
+
+import java.awt.Rectangle
+import java.awt.event.ActionEvent
+import javax.swing.{AbstractAction, JComponent}
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
+
+import org.nlogo.core.Widget
+import org.nlogo.swing.ToolBarButton
+import org.nlogo.window.MouseMode
+
+object WidgetActions {
+
+  val undoManager = new org.nlogo.editor.UndoManager() {
+    // The default one doesn't work!
+    override def canRedo: Boolean = {
+      if(editToBeRedone() != null) {
+        return true
+      }
+      false
```

Comment:
This method can be rewritten as a one-liner. Any time I find myself returning a `true`/`false` from a conditional, I try to think about whether I can make the conditional predicate (`editToBeRedone() != null`, in this case) do the work instead of writing out the entire conditional.